tests/robustness/snap/snap.go

```diff
@@ -0,0 +1,22 @@
+// Package snap describes entities that are capable of performing
+// common snapshot operations
+package snap
+
+// Snapshotter is an interface that describes methods
+// for taking, restoring, deleting snapshots, and
+// tracking them by a string snapshot ID.
+type Snapshotter interface {
```

Comment:
is there a reason why we have all these interfaces? this feels very Java-ish...

Reply:
ultimately the framework can be applied to any backup tool that can implement this interface. I wanted to maintain the flexibility of running experiments and workloads against any arbitrary tool for apples-to-apples comparison against kopia.

Reply:
ok, could it be perhaps separate repo under kopia (longer term, not now)?

Reply:
For sure. This was discussed with @julio-lopez when scoping out the framework. We decided to add it under the rough umbrella of kopia test tools for now, but keeping it independent enough that it can be eventually transitioned to a separate repo (ideally without much overhead to do so).